libs/langchain/langchain/vectorstores/hippo.py

```diff
@@ -0,0 +1,637 @@
+            vectors=[embedding],
+            output_fields=output_fields,
+            topk=k,
+            dsl=expr,
+        )
+        # Organize results.
+        logger.debug(f"[similarity_search_with_score_by_vector] res:{res}")
+        score_col = self._text_field + "%scores"
+        ret = []
+        count = 0
+        for items in zip(*[res[0][field] for field in output_fields]):
+            meta = {field: value for field, value in zip(output_fields, items)}
+            doc = Document(page_content=meta.pop(self._text_field), metadata=meta)
+            logger.debug(
+                f"[similarity_search_with_score_by_vector] "
+                f"res[0][score_col]:{res[0][score_col]}"
+            )
+            score = res[0][score_col][count]
+            count += 1
+            ret.append((doc, score))
+
+        return ret
+
+    @classmethod
+    def from_texts(
+            cls,
+            texts: List[str],
+            embedding: Embeddings,
+            metadatas: Optional[List[dict]] = None,
+            table_name: str = "test",
+            database_name: str = "default",
+            connection_args: Dict[str, Any] = DEFAULT_HIPPO_CONNECTION,
+            index_params: dict = None,
+            search_params=None,
+            drop_old: bool = False,
+            **kwargs: Any,
+    ) -> VectorStore:
+        """
+        Creates an instance of the VST class from the given texts.
+
+        Args:
+            texts (List[str]): List of texts to be added.
+            embedding (Embeddings): Embedding model for the texts.
+            metadatas (List[dict], optional):
+            List of metadata dictionaries for each text.Defaults to None.
+            table_name (str): Name of the table. Defaults to "test".
+            database_name (str): Name of the database. Defaults to "default".
+            connection_args (dict[str, Any]): Connection parameters.
+            Defaults to DEFAULT_HIPPO_CONNECTION.
+            index_params (dict): Indexing parameters. Defaults to None.
+            search_params (dict): Search parameters. Defaults to an empty dictionary.
+            drop_old (bool): Whether to drop the old collection. Defaults to False.
+            kwargs: Other arguments.
+
+        Returns:
+            VST: An instance of the VST class.
+        """
+
+        if search_params is None:
+            search_params = {}
+        logger.info("00 [from_texts] init the class of Hippo")
+        vector_db = cls(
+            embedding_function=embedding,
+            table_name=table_name,
+            database_name=database_name,
+            connection_args=connection_args,
+            index_params=index_params,
+            drop_old=drop_old,
+            **kwargs,
+        )
+        logger.debug(f"[from_texts] texts:{texts}")
+        logger.debug(f"[from_texts] metadatas:{metadatas}")
+        vector_db.add_texts(texts=texts, metadatas=metadatas)
+        return vector_db
+
+# if __name__ == '__main__':
```

Comment:
Is it possible to add unit tests? This PR code is not simple, we need tests. 
After creating ut-s we don't need this commented code anymore.
Also, after creating ut-s, the `Chip War.txt` file should be moved to test folder. Now, we don't keep test data in the production code folders.